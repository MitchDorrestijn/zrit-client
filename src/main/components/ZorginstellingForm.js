import React from 'react';
import {Col, Form, FormGroup, Label, Input, Row, Button} from 'reactstrap';

//this stateless functional component renders the textblock
export default class ZorginstellingForm extends React.Component {

addZorginstelling = () => {
  console.log("api call");
}

  render() {
    return (<div>
      <Form>
        <Row>
        <Col md="4">
          <h5>Plaatsgegevens</h5>
          <FormGroup>
            <Label for="zorginstelling_naam">Naam:</Label>
            <Input type="text" name="zorginstelling_naam" id="zorginstelling_naam" placeholder="Type de naam van de zorginstelling" value=""/>
          </FormGroup>
          <FormGroup>
            <Label for="zorginstelling_adresregel">Adres:</Label>
            <Input type="text" name="zorginstelling_adresregel" id="zorginstelling_adresregel" placeholder="Type het adres van de zorginstelling" value=""/>
          </FormGroup>
          <FormGroup>
            <Label for="zorginstelling_postcode">Postcode:</Label>
            <Input type="text" name="zorginstelling_postcode" id="zorginstelling_postcode" placeholder="Type de postcode van de zorginstelling" value=""/>
          </FormGroup>
          <FormGroup>
            <Label for="zorginstelling_plaats">Plaats:</Label>
            <Input type="text" name="zorginstelling_plaats" id="zorginstelling_plaats" placeholder="Type de plaats van de zorginstelling" value=""/>
          </FormGroup>
        </Col>
        <Col md="4">
          <h5>Accountgegevens</h5>
          <FormGroup>
            <Label for="zorginstelling_email">Email:</Label>
            <Input type="email" name="zorginstelling_email" id="zorginstelling_email" placeholder="Zorginstelling email" value=""/>
          </FormGroup>
          <FormGroup>
            <Label for="zorginstelling_email">Wachtwoord:</Label>
            <Input type="password" name="zorginstelling_wachtwoord" id="zorginstelling_wachtwoord" placeholder="Zorginstelling wachtwoord" value=""/>
          </FormGroup>
        </Col>
      </Row>
      </Form>
      <Button onClick={this.addZorginstelling} color="primary" className='crud-btn'>Toevoegen</Button>
    </div>);
  }
}
