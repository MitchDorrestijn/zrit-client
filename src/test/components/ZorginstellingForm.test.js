import React from 'react';
import {shallow, mount} from 'enzyme';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import ZorginstellingForm from '../../main/components/ZorginstellingForm';
import {Button} from 'reactstrap';

const wrapper = shallow(<ZorginstellingForm/>);

test('<ZorginstellingForm /> should match snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('<ZorginstellingForm /> should have one type of <Form></Form> components', () => {
  expect(wrapper.find('Form').length).toBe(1);
});

test('<ZorginstellingForm /> should have one type of <Row></Row> components', () => {
  expect(wrapper.find('Row').length).toBe(1);
});

test('<ZorginstellingForm /> should render two <Col></Col> components', () => {
  expect(wrapper.find('Col').length).toBe(2);
});

test('<ZorginstellingForm /> should render six <FormGroup></FormGroup> components', () => {
  expect(wrapper.find('FormGroup').length).toBe(6);
});

test('<ZorginstellingForm /> should render six <Input></Input> components', () => {
  expect(wrapper.find('Input').length).toBe(6);
});

test('<ZorginstellingForm /> should render six <Label></Label> components', () => {
  expect(wrapper.find('Label').length).toBe(6);
});

test('<ZorginstellingForm /> calls componentDidMount()', () => {
  sinon.spy(ZorginstellingForm.prototype, 'componentDidMount');
  const wrapper = mount(<ZorginstellingForm/>);
  expect(ZorginstellingForm.prototype.componentDidMount.calledOnce).toBe(true);
});

test('handleAddZorginstelling is called on form submit', () => {
  const handleAddZorginstelling = sinon.spy();
  const wrapper = mount((<Button onClick={handleAddZorginstelling}/>));
  wrapper.find('Button').simulate('click');
  expect(handleAddZorginstelling.calledOnce).toBe(true);
});
