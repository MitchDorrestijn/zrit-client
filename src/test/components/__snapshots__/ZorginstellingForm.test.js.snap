// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZorginstellingForm /> should match snapshot 1`] = `
<div>
  <Form
    tag="form"
  >
    <Row
      tag="div"
    >
      <Col
        md="4"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h5>
          Plaatsgegevens
        </h5>
        <FormGroup
          tag="div"
        >
          <Label
            for="zorginstellingNaam"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Naam:
          </Label>
          <Input
            name="zorginstellingNaam"
            onChange={[Function]}
            placeholder="Type de naam van de zorginstelling"
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Label
            for="zorginstellingAdresregel"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Adres:
          </Label>
          <Input
            name="zorginstellingAdresregel"
            onChange={[Function]}
            placeholder="Type het adres van de zorginstelling"
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Label
            for="zorginstellingPostcode"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Postcode:
          </Label>
          <Input
            name="zorginstellingPostcode"
            onChange={[Function]}
            placeholder="Type de postcode van de zorginstelling"
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Label
            for="zorginstellingPlaats"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Plaats:
          </Label>
          <Input
            name="zorginstellingPlaats"
            onChange={[Function]}
            placeholder="Type de plaats van de zorginstelling"
            type="text"
            value=""
          />
        </FormGroup>
      </Col>
      <Col
        md="4"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h5>
          Accountgegevens
        </h5>
        <FormGroup
          tag="div"
        >
          <Label
            for="zorginstellingEmail"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Email:
          </Label>
          <Input
            name="zorginstellingEmail"
            onChange={[Function]}
            placeholder="Zorginstelling email"
            type="email"
            value=""
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Label
            for="zorginstellingWachtwoord"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Wachtwoord:
          </Label>
          <Input
            name="zorginstellingWachtwoord"
            onChange={[Function]}
            placeholder="Zorginstelling wachtwoord"
            type="password"
            value=""
          />
        </FormGroup>
      </Col>
    </Row>
  </Form>
  <Button
    className="crud-btn"
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    Toevoegen
  </Button>
</div>
`;
